### PR TITLE
Update request-promise-native

### PR DESCRIPTION
**Pull request by robertparkinson**
### Change description

Update dependency and change how we use promises to match the new version.

### Work checklist

- [ ] Unit tests added where applicable
- [ ] Route tests added for new pages
- [ ] New pages included in a11y tests
- [ ] Task list and task completeness checks updated
- [ ] Check and send page updated
- [ ] Routes, page content and page flows of new features are toggled 
- [ ] UI changes look good on mobile
- [ ] Required Google Analytics events are being sent 

### Developer self-QA run statement

- [x] I have clicked through the running application to see if all changes I made actually work.

### Does this PR introduce a breaking change?

- [x] Yes
- [ ] No

### CODE REVIEW

**Comment by andrewwa-kainos** [comment]
Can we pull out common code? The options could be refactored into methods for example

**Comment by markry-kainos** [comment]
@robertparkinson this now passes locally...just missed a couple of the idam functions in new request/promise handling. I'm not sure about the yarn.lock file - why this has soooo many changes 🤔 anyway....bank holiday awaits....

File: src/integration-test/helpers/clients/idamClient.ts [hunk 1]
```diff
@@ -95,49 +92,32 @@ export class IdamClient {
...
-        method: 'POST',
-        uri: `${baseURL}/oauth2/authorize?${upliftParams}`,
-        headers: {
-          Authorization: `Basic ${base64EncodedCredentials}`
-        }
-      }
-      return request(options).then(function (response) {
-        return Promise.resolve()
-      })
+    let code
```
Comment (robertparkinson) [inline src/integration-test/helpers/clients/idamClient.ts]: does this need to be set here? Can't it be a const inside the .then() ?

File: src/integration-test/helpers/clients/idamClient.ts [hunk 2]
```diff
@@ -147,42 +127,27 @@ export class IdamClient {
...
+      },
+      simple: false,
+      followRedirect: false,
+      json: false,
+      resolveWithFullResponse: true
     }
+    return request(options).then(function (response) {
+      return response
+    }).then(function (response) {
+      code = url.parse(response.headers.location, true).query.code
```
Comment (robertparkinson) [inline src/integration-test/helpers/clients/idamClient.ts]: as above

File: src/integration-test/helpers/clients/idamClient.ts (hunk 1 above)
Comment (markry-kainos) [reply src/integration-test/helpers/clients/idamClient.ts]: yeah - i tried this but kept getting tslint errors:
```
error TS2322: Type 'string | string[]' is not assignable to type 'string'.
  Type 'string[]' is not assignable to type 'string'.
```
🤔 have changed to: `const code: any = url.parse(response.headers.location, true).query.code`

**Pull request merged**